onnx/defs/nn/defs.cc

```diff
@@ -1757,6 +1757,104 @@ ONNX_OPERATOR_SET_SCHEMA(
           propagateShapeAndTypeFromFirstInput(ctx);
         }));
 
+
+
+static const char* GroupNormalization_ver11_doc = R"DOC(
+Carries out group normalization as described in the paper
+https://arxiv.org/abs/1803.08494. 
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    GroupNormalization,
+    11,
+    OpSchema()
+        .SetDoc(GroupNormalization_ver11_doc)
+        .Attr(
+            "epsilon",
+            "The epsilon value to use to avoid division by zero.",
+            AttributeProto::FLOAT,
+            1e-5f)
+        .Attr(
+            "num_groups",
+            "The number of groups. It should be greater than 0 and less than or equal to C",
+            AttributeProto::INT)
+        .Input(
+            0,
+            "input",
+           "Input data tensor from the previous operator; "
+            "dimensions are in the form of (N x C x D1 x D2 ... Dn), "
+            "where N is the batch size, C is the number of channels. "
+            "Statistics are computed for every channel of C over N and D1 to Dn dimensions. "
+            "For image data, input dimensions become (N x C x H x W). "
+            "The op also accepts single dimension input of size N in which case C is assumed to be 1",
+            "T")
+        .Input(1, "scale", "The input 1-dimensional scale tensor of size C.", "T")
+        .Input(2, "B", "The input 1-dimensional bias tensor of size C.", "T")
+        .Output(0, "output", "The output tensor of the same shape as input", "T")
+        .TypeConstraint(
+            "T",
+            {"tensor(float16)", "tensor(float)"},
+            // "tensor(double)" - Can not hve tensor(double) because that is not supported by ConstOfShape
+            // But the VerifyTypeConstraint test assumes that the input type is used on the op, which is incorrect
+            "Constrain input and output types to float tensors.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          propagateShapeAndTypeFromFirstInput(ctx);
```

Comment:
Do you have shape inference test?

Reply:
Yes, I have added a shape inference test